src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -578,8 +591,15 @@ private static IConfiguration GetConfiguration(ProjectResource projectResource)
         return configBuilder.Build();
     }
 
-    static bool IsValidAspNetCoreUrl(EndpointAnnotation e) =>
-        e.UriScheme is "http" or "https" && e.TargetPortEnvironmentVariable is null;
+    private static bool ShouldInjectEndpointEnvironment(this IResourceBuilder<ProjectResource> builder, EndpointReference e)
+    {
+        // Check a set of conditions to determine if we should inject environment variables for this endpoint.
+        var endpoint = e.EndpointAnnotation;
+        return
+            endpoint.UriScheme is "http" or "https" && // Only process http and https endpoints
+            endpoint.TargetPortEnvironmentVariable is null && // Skip if target port env variable was set
```

Comment:
Do we keep these or are these in the default filter?

Reply:
Yes, they could go in the default filter, but that then makes an even stronger case for doing the transparent delegation, per other comment. Otherwise, if a user chooses not to delegate, they'll lose those default filter, which is very likely not what they want.

Reply:
OK I can buy that.

Reply:
Done as part of the composition change.